vcr/deploy/deploy.go

```diff
@@ -431,3 +434,12 @@ func Deploy(ctx context.Context, opts *Options, createPkgResp api.CreatePackageR
 
 	return deploymentResponse, nil
 }
+
+func hasReadAndWritePermission(path string) bool {
+	file, err := os.OpenFile(path, os.O_RDWR, 0666)
+	if err != nil {
+		return false
+	}
+	defer file.Close()
+	return true
+}
```

Comment:
instead of checking if file has read write permissions, just check the error is a permissions error when it can't compress it. it might slow down the compression too much if you check every file